packages/truffle-interface-adapter/lib/web3-shim.js

```diff
@@ -0,0 +1,33 @@
+const Web3 = require("web3");
+const BN = require("bn.js");
+
+// This is a temporary shim to support the basic issues with Quorum
+
+class Web3Shim {
+  constructor(options) {
+    let web3 = new Web3();
+
+    if (options.provider) {
+      web3.setProvider(options.provider);
+    }
+
+    if (options.quorum) {
+      // duck punch the block output formatter since quorum uses nanoseconds in the timestamp
+      // field instead of seconds
+      const _oldFormatter = web3.eth.getBlock.method.outputFormatter;
+      web3.eth.getBlock.method.outputFormatter = block => {
+        const _oldTimestamp = block.timestamp;
+        let timestamp = new BN(block.timestamp.slice(2), 16);
+        timestamp = timestamp.div(new BN(10).pow(new BN(9)));
+        block.timestamp = "0x" + timestamp.toString(16);
+        let result = _oldFormatter.call(web3.eth.getBlock.method, block);
+        result.timestamp = _oldTimestamp;
+        return result;
+      };
+    }
+
+    return web3;
+  }
+}
```

Comment:
I love me a good shim. So much utility! It's beautiful.

![shim](https://user-images.githubusercontent.com/151065/53857016-a3fc8d00-3fa2-11e9-8e27-064337603c21.jpg)